README.md

```diff
@@ -8,3 +8,33 @@ Currently implemented applications are
     - Full-service cross-chain message delivery application that is configurable to listen to specific source and destination chain pairs and relay messages according to its configured rules.
 2. [Signature Aggregator](signature-aggregator/README.md)
     - Lightweight API that requests and aggregates signatures from validators for any ICM message, and returns a valid signed message that the user can then self-deliver to the intended destination chain.
+
+## Updating dependencies and E2E testing
+
+Applications in this repository depend on the following upstream repositories, both directly in terms of code imports defined in `go.mod` file as well as indirectly for E2E tests where binary versions are used to spin up the test network via `tmpnet`:
+
+1.  [avalanchego](https://github.com/ava-labs/avalanchego/)
+2.  [coreth](https://github.com/ava-labs/coreth) (indirectly)
+3.  [subnet-evm](https://github.com/ava-labs/subnet-evm)
+
+> [!NOTE]
+> We require any commits referenced in our `main` branch to be present in the default branches of the repositories above, but during active development it might be useful to work against changes in progress that are still on feature branches.
+
+When developing such features that require updates to one or more of the above, care must be taken to understand where the relevant code comes from. The binaries of applications built in this repo are built against versions referenced in the `go.mod` file. The E2E tests run against a simulated network running locally that is started by calling a separately compiled `avalanchego` binary as well as its plugins. These are compiled based on the values of `AVALANCHEGO_VERSION` in the local checkout of `subnet-evm` when running the tests locally and directly in this repository's `./scripts/versions.sh` when running E2E tests remotely through GitHub actions. 
+
+`avalanchego` and `coreth` have a direct circular dependency and this repository is only indirectly dependent on `coreth` but directly on `avalanchego`. Therefore if any updates are required from the `coreth` side, a corresponding `avalanchego` commit referencing those changes is required. On the other hand `subnet-evm` just depends directly on `avalanchego`.
+
+The most complicated example case that can arise above is that a feature depends on a new change in `coreth`. In this case, the developer needs to:
```

Comment:
Looks like this section is incomplete.

Reply:
The cases below this actually explain the workflow that I'm introducing here. I can make them bold instead of headings to make it clearer perhaps?

Reply:
I think it's just the wording, usually when using : i expect a list to follow right after

Reply:
"in this case the developer would need to perform the following steps for either remote or local e2e testing cases." or similar

Reply:
Yeah, the separation between headings is confusing. Perhaps explicitly mention the headings, or mention the "following sections"